libnavui-tripprogress/src/main/java/com/mapbox/navigation/ui/tripprogress/view/MapboxTripProgressView.kt

```diff
@@ -116,6 +119,38 @@ class MapboxTripProgressView : FrameLayout {
         )
     }
 
+    /**
+     * Applies the necessary view side effects based on the input.
+     *
+     * If [NavigationRoute.directionsRoute] contains only one [RouteLeg] then `totalTime`,
+     * `totalDistance` and `totalEstimatedTimeToArrival` will be equal to `legTime`, `legDistance`
+     * and `estimatedTimeToArrival` for the only [RouteLeg].
+     *
+     * If [NavigationRoute.directionsRoute] contains more than one [RouteLeg] then `totalTime`,
+     * `totalDistance` and `totalEstimatedTimeToArrival` will be equal to the sum of `legTime`,
+     * `legDistance` and `estimatedTimeToArrival` for the each [RouteLeg].
```

Comment:
```suggestion
     * Draws `totalTime`, totalDistance` and `totalEstimatedTimeToArrival` based on the total time of the route (including all upcoming legs).
```

We draw always based on the whole route, no need to be so granular in the docs.

On that note, shouldn't we draw ETA for the current leg? Our Nav Camera only displays the current leg in overview by default, the progress update API does the same.

Reply:
I thought about that quite a lot. Not only the ETA, but duration and distance as well. Should we show it for the entire route or the current leg? This might be debatable as there might be various different opinions among the users. We can keep this API as is and introduce another API `renderTripDetail(routeLeg: RouteLeg, result: TripDetailsValue)` which will render the details for the `leg` passed in as input. Wdyt?

@LukasPaczos